projects/plugins/jetpack/extensions/shared/components/product-management-controls/toolbar-control.js

```diff
@@ -65,11 +68,16 @@ function Product( { onClose, product, selectedProductId, setSelectedProductId }
 	);
 }
 
-function NewProduct( { onClose, productType } ) {
+function NewProduct( { onClose } ) {
+	const { clientId, productType } = useProductManagementContext();
 	const openBlockSidebar = useOpenBlockSidebar();
+	const { selectBlock } = useDispatch( blockEditorStore );
 
 	const handleClick = event => {
 		event.preventDefault();
+		if ( clientId ) {
+			selectBlock( clientId );
+		}
```

Comment:
Could we add this check and the selectBlock inside the openBlockSidebar() call? Maybe we could invoke it like:
````
openBlockSidebar( clientId );
````
And let use-open-block-sidebar.js handle these checks, what do you think?